File: 240-search-a-2d-matrix-ii.go
func searchMatrix(matrix [][]int, target int) bool {
    if len(matrix) == 0 || len(matrix[0]) == 0{
        return false
    }
    row := 0
    col := len(matrix[0]) - 1
    for col >= 0 && row < len(matrix){
        v := matrix[row][col]
        if v == target{
            return true
        } else if v > target {
            col -= 1
        } else {
            row += 1
        }
        
    }
    return false
}

File: bwu/array/240-search-a-2d-matrix-ii.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if not len(matrix) or not len(matrix[0]):
            return False
            
        m, n = len(matrix), len(matrix[0])
        x, y = m - 1, 0
        while x >= 0 and y < n:
            if matrix[x][y] == target:
                return True
            elif matrix[x][y] > target:
                x -= 1
            else:
                y += 1
        return False
